content/courses/advanced-ada/chapters/arrays.rst

```diff
@@ -31,6 +31,369 @@ Unconstrained Arrays
+array type called :ada:`Measurements` to store this data. Also, we can have
+three procedures for this array:
+
+- :ada:`Show_Indices`, which presents the indices (days and hours) of the
+  two-dimensional array;
+
+- :ada:`Show_Values`, which presents the values stored in the array; and
+
+- :ada:`Reset`, which resets each value of the array.
+
+This is the complete code for this application:
+
+.. code:: ada run_button project=Courses.Advanced_Ada.Arrays.Multidimensional_Measurements
+
+    package Measurement_Defs is
+
+       type Days is (Mon, Tue, Wed, Thu, Fri, Sat, Sun);
+
+       type Hours is range 0 .. 11;
+
+       subtype Measurement is Float;
+
+       type Measurements is array (Days, Hours) of Measurement;
+
+       procedure Show_Indices (M : Measurements);
+
+       procedure Show_Values (M : Measurements);
+
+       procedure Reset (M : out Measurements);
+
+    end Measurement_Defs;
+
+    with Ada.Text_IO;      use Ada.Text_IO;
+
+    package body Measurement_Defs is
+
+       procedure Show_Indices (M : Measurements) is
+       begin
+          Put_Line ("---- Indices ----");
+
+          for D in M'Range (1) loop
+             Put (D'Image & " ");
+
+             for H in M'First (2) .. M'Last (2) - 1 loop
+                Put (H'Image & " ");
+             end loop;
+             Put_Line (M'Last (2)'Image);
+          end loop;
+       end Show_Indices;
+
+       procedure Show_Values (M : Measurements) is
+          package H_IO is
+            new Ada.Text_IO.Integer_IO (Hours);
+          package M_IO is
+            new Ada.Text_IO.Float_IO (Measurement);
+
+            procedure Set_IO_Defaults is
+            begin
+               H_IO.Default_Width := 5;
+
+               M_IO.Default_Fore  := 1;
+               M_IO.Default_Aft   := 2;
+               M_IO.Default_Exp   := 0;
+            end Set_IO_Defaults;
+       begin
+          Set_IO_Defaults;
+
+          Put_Line ("---- Values ----");
+          Put ("   ");
+          for H in M'Range (2) loop
+             H_IO.Put (H);
+          end loop;
+          New_Line;
+
+          for D in M'Range (1) loop
+             Put (D'Image & " ");
+
+             for H in M'Range (2) loop
+                M_IO.Put (M (D, H));
+                Put (" ");
+             end loop;
+             New_Line;
+          end loop;
+       end Show_Values;
+
+       procedure Reset (M : out Measurements) is
+       begin
+          M := (others => (others => 0.0));
+       end Reset;
+
+    end Measurement_Defs;
+
+    with Measurement_Defs; use Measurement_Defs;
+
+    procedure Show_Measurements is
+       M : Measurements;
+    begin
+       Reset (M);
+       Show_Indices (M);
+       Show_Values (M);
+    end Show_Measurements;
+
+We recommend that you spend some time analyzing this example. Also, we'd like
+to highlight the following aspects:
+
+- We access a value from a multidimensional array by using commas. For example:
```

Comment:
"We access a value from a multidimensional array by using commas." =>
"We access a value from a multidimensional array by using commas to separate the separate index values within the parentheses."

(or something like that)

Reply:
Your version is much better, thanks!